python/paddle/fluid/compiler.py

```diff
@@ -499,6 +499,175 @@ def _get_places(self, place, place_list):
         return place_list
 
 
+class IpuDynamicPatcher(object):
+    """
+    Patcher for IPU dynamic2static support.
+    """
+
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def convert_concrete_program(ipu_strategy,
+                                 concrete_program,
+                                 class_instance=None):
+        """
+        Convert the ConcreteProgram to IPUConcreteProgram.
+        """
+        from ..fluid.dygraph.base import switch_to_static_graph
+        from ..fluid import backward
+        from ..fluid.initializer import Constant
+        from ..fluid.framework import device_guard
+        import paddle
```

Comment:
Recommend to add import statement in front of file.

Reply:
unfortunately, move those imports at front introduced import loop issue. We may need to keep it here.